packages/plugin-json/src/lib/amplience-schema-transformers.ts

```diff
@@ -115,6 +122,34 @@ export const objectProperties = (
       ]) ?? []
   );
 
+const getDeliveryKeyDirective = (field?: FieldDefinitionNode) => field && maybeDirective(field, 'amplienceDeliveryKey')
+const hasDeliveryKeyDirective = (field: FieldDefinitionNode) => !!maybeDirective(field, 'amplienceDeliveryKey')
+
+const deliveryKeyMetaProperty = (type: ObjectTypeDefinitionNode): AmplienceDeliveryKeyType | undefined => {
```

Comment:
This function should not be allowed to return undefined, since undefined is not a property (your function name says this returns a "Property").
Instead guarantee outside of this function that there is a delivery key directive and make this function just a pure data builder that always returns an object. The directive should probably be part of the input for this function.

Reply:
Done. I did a little additional refactoring:

- definitite `deliveryKeyDirective` as param to `deliveryKeyMetaProperty` -> always returns an `AmplienceDeliveryKeyType`
- `properties` and `propertyOrder` now manipulated if directive present at the top of `contentTypeSchemaBody`
- `getDeliveryKeyDirective`:
    - renamed to `maybeDeliveryKeyDirective` for consistency with underlying `maybeDirective`
    - now finds the appropriate field internally rather than relying on an external filter for an optional `field` param